dotCMS/src/main/java/com/dotcms/services/VanityUrlServices.java

```diff
@@ -132,27 +122,77 @@ public void updateCache(CachedVanityUrl vanity) {
     }
 
     /**
-     * Get the cached vanity Url from the primary cache
+     * Get the cached vanity Url from the primary cache for a given site and SYSTEM_HOST
      *
      * @param uri The current uri
      * @param siteId The current site Id
      * @param languageId The current language Id
      * @return CachedVanityUrl object
      */
     public CachedVanityUrl getCachedVanityUrlByUri(String uri, String siteId, long languageId) {
-        return vanityURLCache.get(VanityUrlUtil.sanitizeKey(siteId, uri, languageId));
+
+        CachedVanityUrl foundVanity;
+        if (null != siteId && !siteId.equals(Host.SYSTEM_HOST)) {
+
+            //First search in cache with the given site
+            foundVanity = vanityURLCache.get(VanityUrlUtil.sanitizeKey(siteId, uri, languageId));
+
+            //If nothing found lets try with the SYSTEM_HOST
+            if (null == foundVanity) {
+                foundVanity = vanityURLCache
+                        .get(VanityUrlUtil.sanitizeKey(Host.SYSTEM_HOST, uri, languageId));
+            }
+        } else {
+            foundVanity = vanityURLCache
+                    .get(VanityUrlUtil.sanitizeKey(Host.SYSTEM_HOST, uri, languageId));
+        }
+
+        return foundVanity;
     }
 
     /**
-     * Get the list of cached Vanity Url associated to a site
+     * Get the list of cached Vanity URLs associated to a given site and SYSTEM_HOST
      *
      * @param siteId The current site Id
      * @param languageId The current language Id
      * @return A set of CachedVanityUrl
      */
     public Set<CachedVanityUrl> getVanityUrlBySiteAndLanguage(String siteId, long languageId) {
-        return CacheLocator.getVanityURLCache()
-                .getCachedVanityUrls(VanityUrlUtil.sanitizeSecondCacheKey(siteId, languageId));
+
+        Set<CachedVanityUrl> foundVanities;
+        if (null != siteId && !siteId.equals(Host.SYSTEM_HOST)) {
+
+            //First search in cache with the given site id
+            foundVanities = CacheLocator.getVanityURLCache()
+                    .getCachedVanityUrls(VanityUrlUtil.sanitizeSecondCacheKey(siteId, languageId));
+
+            //null means we need to initialize the cache for this site
+            if (null == foundVanities) {
+                return foundVanities;
+            }
+
+            //Now search in cache with the SYSTEM_HOST
+            Set<CachedVanityUrl> systemHostFoundVanities = CacheLocator.getVanityURLCache()
+                    .getCachedVanityUrls(
+                            VanityUrlUtil.sanitizeSecondCacheKey(Host.SYSTEM_HOST, languageId));
+
+            if (null != systemHostFoundVanities) {
+                if (null != foundVanities) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)